### PR TITLE
[Fix] Make sure `mark.size` are correctly redirected for bar, text, and tick

### PR DESCRIPTION
**Pull request by kanitw**

### CODE REVIEW

**Comment by kanitw** [comment]
Please manually merge to master and push to avoid unnecessary rebases in dependent branches.

File: src/compile/mark/mixins.ts [hunk 1]
```diff
@@ -34,19 +34,21 @@ export function color(model: UnitModel) {
   return e;
 }
 
-export function baseEncodeEntry(model: UnitModel, ignoreOrient: boolean) {
+export type Ignore = Record<'size' | 'orient', 'ignore' | 'include'>;
```
Comment (domoritz) [inline src/compile/mark/mixins.ts]: Didn't know about this one.

File: src/compile/mark/mixins.ts [hunk 2]
```diff
@@ -121,7 +123,7 @@ export function bandPosition(fieldDef: FieldDef<string>, channel: 'x'|'y', model
   const scaleName = model.scaleName(channel);
   const sizeChannel = channel === 'x' ? 'width' : 'height';
 
-  if (model.encoding.size) {
+  if (model.encoding.size || model.markDef.size !== undefined) {
```
Comment (domoritz) [inline src/compile/mark/mixins.ts]: Why are you just checking for `model.encoding.size` and not `model.encoding.size !== undefined`?

File: test/compile/mark/tick.test.ts [hunk 3]
```diff
@@ -161,7 +177,7 @@ describe('Mark: Tick', function() {
       'data': {'url': 'data/cars.json'},
     });
     const props = tick.encodeEntry(model);
-    it('height should change with size field', function() {
+    it('maps height ot size', function() {
```
Comment (domoritz) [inline test/compile/mark/tick.test.ts]: ot -> not

File: src/compile/mark/mixins.ts (hunk 2 above)
Comment (kanitw) [reply src/compile/mark/mixins.ts]: `markDef.size` is number so `0` means something

`encoding.size` can only be an object.

File: test/compile/mark/tick.test.ts (hunk 3 above)
Comment (kanitw) [reply test/compile/mark/tick.test.ts]: lol it's actually "to"

File: src/compile/mark/mixins.ts (hunk 2 above)
Comment (domoritz) [reply src/compile/mark/mixins.ts]: I see. I'd be okay with checking against undefined in both cases.

**Pull request merged**